app/views/tutorial/a1.html

```diff
@@ -0,0 +1,409 @@
+                                <code>setInterval()</code>, and
+                                <code>Function()</code>.
+                            </li>
+                            <li>
+                                For parsing JSON input, instead of using
+                                <code>eval()</code>, use a safer alternative such as
+                                <code>JSON.parse()</code>. For type conversions use type related
+                                <code>parseXXX()</code>methods.
+                            </li>
+                            <li>Include
+                                <code>"use strict"</code>at the beginning of a function, which enables <a target="_blank" href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions_and_function_scope/Strict_mode"> strict mode </a>within the enclosing function scope.</li>
+
+                        </ul>
+                    </div>
+                </div>
+                <div class="panel panel-default">
+                    <div class="panel-heading">
+                        <h3 class="panel-title">Source Code Example</h3>
+                    </div>
+                    <div class="panel-body">
+                        <p>In
+                            <code>routes/contributions.js</code>, the
+                            <code>handleContributionsUpdate()</code>function insecurely uses
+                            <code>eval()</code>to convert user supplied contribution amounts to integer.
+                            <pre>
+        // Insecure use of eval() to parse inputs
+        var preTax = eval(req.body.preTax);
+        var afterTax = eval(req.body.afterTax);
+        var roth = eval(req.body.roth);
+                            </pre> This makes application vulnerable to SSJS attack. It can fixed simply by using
+                            <code>parseInt()</code>instead.
+                            <pre>
+        //Fix for A1 -1 SSJS Injection attacks - uses alternate method to eval
+        var preTax = parseInt(req.body.preTax);
+        var afterTax = parseInt(req.body.afterTax);
+        var roth = parseInt(req.body.roth);
+                            </pre>
+                        </p>
+                        <p>In addition, all functions begin with
+                            <code>use strict</code>pragma.
+                    </div>
+                </div>
+                <div class="panel panel-default">
+                    <div class="panel-heading">
+                        <h3 class="panel-title">Further Reading</h3>
+                    </div>
+                    <div class="panel-body">
+                        <ul>
+                            <li><a target="_blank" href="https://media.blackhat.com/bh-us-11/Sullivan/BH_US_11_Sullivan_Server_Side_WP.pdf">“ServerSide JavaScript Injection: Attacking NoSQL and Node.js"</a> a whitepaper by Bryan Sullivan.</li>
```

Comment:
#### Automated PR Comment From Polaris SAST
<!-- 15aa0bbaf8edc3af7c0599cf853a2e25 -->
<!-- Comment managed by Git Commenter, do not modify! -->
### Polaris SAST Issue - Insecure Link Target
_Low CWE-1022_
The application uses an anchor tag whose `target` attribute is set to `_blank`, but the `rel` attribute is not set with the `noopener` value. The page opened from that target link may have the access to the `location` object of the original page through `window.opener.location`. A malicious page associated with this link will be able to navigate the original page to an arbitrary site, such as a phishing site, and thus, execute a reverse tabnabbing attack.
Note that most contemporary browsers add the `noopener` value to the `rel` automatically by default. Thus, only users with older versions of browsers are targets for this phishing attack.



### How to fix
To prevent untrusted pages linked from the application from being able to manipulate the "window.opener" object, set the attribute rel="noopener noreferrer" for each HTML link to external sites. In addition to preventing reverse tabnabbing attacks, it may also improve browser performance as the new page can be run in a separate process.

When the rel attribute is set to "noopener", the "window.opener" object is "null" on the new page. Therefore, the attacker cannot manipulate the navigation of the original page.  Additionally, the "noreferrer" attribute tells the browser to not include referrer information when the new page is requested.

For links opened with JavaScript, add the values "noopener", "noreferrer" as the "windowFeatures" parameter in the "window.open()" function:
```
var childWindow = window.open(url, name, "noopener,noreferrer");   
```
For legacy browser support, you can also manually set the opener to "null":
```
childWindow.opener = null;  
```

**Note**: The rel="noopener" attribute setting is supported by most modern browsers such as Firefox, Chrome, Edge and Safari but is not supported by legacy browsers such as Internet Explorer. For the latest information on browser support, refer to https://caniuse.com/#search=rel.